presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -506,6 +509,9 @@ public IndexMetadata getIndexMetadata(String index)
             catch (IOException e) {
                 throw new PrestoException(ELASTICSEARCH_INVALID_RESPONSE, e);
             }
+            catch (NoSuchElementException e) {
+                throw new PrestoException(ELASTICSEARCH_QUERY_FAILURE, "SELECT * not allowed from relation that has no columns");
```

Comment:
Would this exception be better here? Since this can only occur if there are no JsonNode returned from `OBJECT_MAPPER.readTree(body)`

```suggestion
                throw new PrestoException(ELASTICSEARCH_INVALID_RESPONSE, "No mappings found for index: " + index");
```

Reply:
Done